bench/bench_bounded_stack.ml

```diff
@@ -0,0 +1,72 @@
+open Multicore_bench
+module Stack = Saturn_lockfree.Bounded_stack
+
+let run_one_domain ~budgetf ?(n_msgs = 50 * Util.iter_factor) () =
+  let t = Stack.create () in
+
+  let op push = if push then Stack.push t 101 else Stack.pop_opt t |> ignore in
+
+  let init _ =
+    assert (Stack.is_empty t);
+    Util.generate_push_and_pop_sequence n_msgs
+  in
+  let work _ bits = Util.Bits.iter op bits in
+
+  Times.record ~budgetf ~n_domains:1 ~init ~work ()
+  |> Times.to_thruput_metrics ~n:n_msgs ~singular:"message" ~config:"one domain"
+
+let run_one ~budgetf ?(n_adders = 2) ?(n_takers = 2)
+    ?(n_msgs = 50 * Util.iter_factor) () =
+  let n_domains = n_adders + n_takers in
+
+  let t = Stack.create () in
+
+  let n_msgs_to_take = Atomic.make 0 |> Multicore_magic.copy_as_padded in
+  let n_msgs_to_add = Atomic.make 0 |> Multicore_magic.copy_as_padded in
```

Comment:
Multicore-bench now has a [`Countdown` abstraction](https://ocaml-multicore.github.io/multicore-bench/doc/multicore-bench/Multicore_bench/Countdown/index.html) for this.